Tests/MongoSwiftTests/DocumentTests.swift

```diff
@@ -372,4 +373,101 @@ final class DocumentTests: XCTestCase {
         expect(doc["a1"]).to(equal(arr1))
         expect(doc["a2"]).to(equal(arr2))
     }
+
+    func testSetValueForExistingKey() throws {
+        let overwritables: Document = ["double": 2.5, "int32": Int32(32), "int64": Int64(64), "bool": false]
+        let oid = ObjectId()
+        let nonOverwritables: Document = ["string": "hello", "nil": nil, "doc": ["x": 1] as Document,
+                                          "arr": [1, 2] as [Int], "oid": oid]
+
+        // test replacing `Overwritable` types with values of their own type
+        do {
+            var doc = overwritables
+
+            // overwrite int32 with int32
+            doc["int32"] = Int32(15)
+            expect(doc["int32"] as? Int).to(equal(15))
+
+            // overwrite int32 with int
+            doc["int32"] = 20
+            expect(doc["int32"] as? Int).to(equal(20))
+
+            // overwrite int64 with int64
+            let bigInt64 = Int64(Int32.max) + 1
```

Comment:
Do we need to skip some of these "big int" tests when running on 32-bit platforms? I seem to recall another test doing that (perhaps with macros) when working with large integers.

Reply:
good point. I've moved your code from that test to be in an `XCTestCase` extension so we can access `is32Bit` from any test. I have refactored this test so that we use a different version of `overwritables` not containing an `Int64` on 32-bit platforms.

however, there are most definitely other cases throughout our older tests where we are not performing this check, so I've opened [SWIFT-197](https://jira.mongodb.org/browse/SWIFT-197) so we remember to do an audit in the future.